xarray/tests/test_dataarray.py

```diff
@@ -2077,6 +2077,15 @@ def test_upsample_interpolate(self):
             # done here due to floating point arithmetic
             self.assertDataArrayAllClose(expected, actual, rtol=1e-16)
 
+    @requires_scipy
+    def test_upsample_interpolate_regression_1605(self):
+        dates = pd.date_range('2016-01-01', '2016-03-31', freq='1D')
+        expected = xr.DataArray(np.random.random((len(dates), 2, 3)),
+                                dims=('time', 'x', 'y'),
+                                coords={'time': dates})
+        actual = expected.resample(time='1D').interpolate('linear')
+        self.assertDataArrayAllClose(actual, expected, rtol=1e-16)
```

Comment:
@jhamman reminder to use the assert functions rather than `self. ...` methods, for future PRs